app/src/lib/stores/pull-request-store.ts

```diff
@@ -48,25 +50,48 @@ export class PullRequestStore {
 
       await this.writePRs(raw, repository)
 
-      const results = await this.getPullRequests(repository)
-
-      await this.getStatusForPRs(results, repository, account)
-
-      this.pullRequests = await this.getPullRequests(repository)
-      this.emitUpdate()
+      const prs = await this.getPullRequests(repository)
+      await this.refreshStatusForPRs(prs, repository, account)
     } catch (error) {
       log.warn(`Error refreshing pull requests for '${repository.name}'`, error)
       this.emitError(error)
+    } finally {
+      this.changeActiveFetchCount(repository, c => c - 1)
+      this.emitUpdate(repository)
     }
   }
 
+  private changeActiveFetchCount(
+    repository: GitHubRepository,
+    fn: (count: number) => number
+  ) {
+    const key = forceUnwrap(
+      'Cannot fetch PRs for a repository which is not in the database',
+      repository.dbID
+    )
+    const currentCount = this.activeFetchCountPerRepository.get(key) || 0
+    const newCount = fn(currentCount)
+    this.activeFetchCountPerRepository.set(key, newCount)
+  }
+
+  /** Is the store currently fetching the list of open pull requests? */
+  public isFetchingPullRequests(repository: GitHubRepository): boolean {
+    const key = repository.dbID
```

Comment:
We are using `forceUnwrap` above, is there a reason not to use it here?

Reply:
I was less certain that we could make that guarantee for this call... but thinking about it more, I _think_ we can 👍